pandas/core/arrays/categorical.py

```diff
@@ -1366,7 +1366,10 @@ def remove_categories(self, removals) -> Self:
             removals = [removals]
 
         removals = Index(removals).unique().dropna()
-        new_categories = self.dtype.categories.difference(removals)
```

Comment:
Can you just replace the `if/else` with `difference(removals, sort=not self.dtype.ordered)`?